src/Core/OnExceptionDispatchers/OnExceptionSkipOrExitDispatcher.cs

```diff
@@ -0,0 +1,52 @@
+﻿// <copyright file="OnExceptionSkipOrExitDispatcher.cs" company="Cognisant">
+// Copyright (c) Cognisant. All rights reserved.
+// </copyright>
+
+namespace CR.MessageDispatch.Core
+{
+    using System;
+
+    /// <summary>
+    /// Sets up a dispatcher which will either skip or exit when an exception is thrown.
+    /// </summary>
+    /// <typeparam name="TMessage">Message Type</typeparam>
+    public class OnExceptionSkipOrExitDispatcher<TMessage> : IDispatcher<TMessage>
+    {
+        private readonly IDispatcher<TMessage> _dispatcher;
+        private readonly Func<TMessage, Exception, bool> _shouldExit;
+        private readonly Action<TMessage, Exception> _onExitLog;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="OnExceptionSkipOrExitDispatcher{TMessage}"/> class.
+        /// </summary>
+        /// <param name="dispatcher">Dispatcher to use.</param>
```

Comment:
These need better descriptions. Something like the following maybe?

"The inner dispatcher which this dispatcher should wrap"
"A `Func` which decides what to do when an exception is thrown by the inner dispatcher. It should return `false` to instruct the dispatcher to skip the current message and continue processing future messages, and `true` if error is fatal and the application should exit.
"the onExitLog callback will be called prior to the application exiting in case of a fatal error (as indicated by `shouldExit`). This gives an opportunity to log the exception.